crates/uv/src/commands/python/install.rs

```diff
@@ -291,42 +292,102 @@ pub(crate) async fn install(
+                                    to.user_display()
+                                ),
+                            ));
+                            continue;
+                        }
+                        debug!(
+                            "Replacing existing executable at {} due to `--force`",
+                            target.user_display()
+                        );
+                    }
+                    Some(existing) if existing == installation => {
+                        // The existing link points to the same installation, so we're done unless
+                        // they requested we reinstall
+                        if !(reinstall || force) {
+                            debug!(
+                                "Executable at {} is already for {}",
+                                target.user_display(),
+                                installation.key(),
+                            );
+                            continue;
+                        }
+                        debug!(
+                            "Replacing existing executable for {} at {}",
                             installation.key(),
-                            anyhow::anyhow!(
-                                "Executable already exists at `{}`. Use `--reinstall` to force replacement.",
-                                to.user_display()
-                            ),
-                        ));
+                            target.user_display(),
+                        );
+                    }
+                    Some(existing) => {
+                        // The existing link points to a different installation, check if it
+                        // is reasonable to replace
+                        if force {
+                            debug!(
+                                "Replacing existing executable for {} at {} with executable for {} due to `--force` flag",
+                                existing.key(),
+                                target.user_display(),
+                                installation.key(),
+                            );
+                        } else {
+                            if installation.is_upgrade_of(existing) {
+                                debug!(
+                                    "Replacing existing executable for {} at {} with executable for {} since it is an upgrade",
+                                    existing.key(),
+                                    target.user_display(),
+                                    installation.key(),
+                                );
+                            } else {
+                                debug!(
+                                    "Executable already exists at `{}` for `{}`. Use `--force` to replace.",
+                                    existing.key(),
+                                    to.user_display()
+                                );
```

Comment:
Should any of this be user-facing? I'm unsure... At least this last one seems like it should be, though? Otherwise we risk silent no-ops here.

Reply:
I'm not sure either. I opted for this to be silent because if I ran `uv python install 3.12.6` then `uv python install 3.12.4` I would not expect the `python3.12` executable to change.